File: 001-010/010/010.py
#!/usr/bin/env python


def isPrime(number, primes):
    for prime in primes:
        if number**0.5 >= prime:
            if number % prime == 0:
                return 1
        else:
            return 0


def main():
    total = 2
    primes = [2]
    for x in range(3, 2000000, 2):
        if isPrime(x, primes) == 0:
            primes.append(x)
            total += x
    print(total)


if __name__ == "__main__":
    main()

File: 001-010/010/010.rb
#!/usr/local/bin/ruby -w

def prime?(number, primes)
    primes.each do |prime|
        if number**0.5 >= prime
            return 1 if (number % prime).zero?
        else
            return 0
        end
    end
end


total = 2
primes = [2]
(3..2_000_000).step(2).each do |x|
    if prime?(x, primes).zero?
        primes.push(x)
        total += x
    end
end

puts total
